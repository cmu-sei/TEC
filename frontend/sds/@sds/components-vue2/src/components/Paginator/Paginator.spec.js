// TEC - Automated Mismatch Detection Tool
// Copyright 2023 Carnegie Mellon University.
// NO WARRANTY. THIS CARNEGIE MELLON UNIVERSITY AND SOFTWARE ENGINEERING INSTITUTE MATERIAL IS FURNISHED ON AN "AS-IS" BASIS. CARNEGIE MELLON UNIVERSITY MAKES NO WARRANTIES OF ANY KIND, EITHER EXPRESSED OR IMPLIED, AS TO ANY MATTER INCLUDING, BUT NOT LIMITED TO, WARRANTY OF FITNESS FOR PURPOSE OR MERCHANTABILITY, EXCLUSIVITY, OR RESULTS OBTAINED FROM USE OF THE MATERIAL. CARNEGIE MELLON UNIVERSITY DOES NOT MAKE ANY WARRANTY OF ANY KIND WITH RESPECT TO FREEDOM FROM PATENT, TRADEMARK, OR COPYRIGHT INFRINGEMENT.
// Released under a BSD (SEI)-style license, please see license.txt or contact permission@sei.cmu.edu for full terms.
// [DISTRIBUTION STATEMENT A] This material has been approved for public release and unlimited distribution.  Please see Copyright notice for non-US Government use and distribution.
// This Software was created using the SEI Design System and includes and/or makes use of Third-Party Software subject to its own license.
// DM23-003
 
 
import { shallowMount } from '@vue/test-utils'
import Component from "./Paginator.vue";

describe("Paginator.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });

  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with totalPages prop assigned", () => {
    const propsData = {
      totalPages: 10,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with currentPage and totalPages prop assigned", () => {
    const propsData = {
      currentPage: 3,
      totalPages: 10,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with loading and totalPages prop assigned", () => {
    const propsData = {
      loading: true,
      totalPages: 10,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with threshold and totalPages prop assigned", () => {
    const propsData = {
      threshold: 10,
      totalPages: 100,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with currentPage, threshold, loading, and totalPages prop assigned", () => {
    const propsData = {
      currentPage: 20,
      threshold: 10,
      loading: true,
      totalPages: 100,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot when threshold is >= totalPages prop", () => {
    const propsData = {
      threshold: 10,
      totalPages: 3,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot when currentPage > totalPages - threshold + 1", () => {
    const propsData = {
      currentPage: 12,
      threshold: 5,
      totalPages: 15,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("fires the goToPage method when a button is clicked", () => {
    const spy = jest.spyOn(Component.methods, "goToPage");
    const propsData = {
      currentPage: 4,
      totalPages: 15,
    };
    const wrapper = shallowMount(Component, { propsData });
    wrapper.find("button").trigger("click");
    expect(spy).toHaveBeenCalled();
    expect(wrapper.emitted("go-to-page")).toBeTruthy();
  });
});
