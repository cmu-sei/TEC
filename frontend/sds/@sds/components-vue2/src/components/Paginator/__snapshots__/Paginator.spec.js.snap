// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator.vue matches snapshot when currentPage > totalPages - threshold + 1 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button class=\\"btn btn-default\\">
        1
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button class=\\"btn btn-default\\">
        11
      </button><button class=\\"btn btn-default active\\">
        12
      </button><button class=\\"btn btn-default\\">
        13
      </button><button class=\\"btn btn-default\\">
        14
      </button><button class=\\"btn btn-default\\">
        15
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 12
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;

exports[`Paginator.vue matches snapshot when threshold is >= totalPages prop 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button disabled=\\"disabled\\" title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button class=\\"btn btn-default active\\">
        1
      </button><button class=\\"btn btn-default\\">
        2
      </button><button class=\\"btn btn-default\\">
        3
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 1
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;

exports[`Paginator.vue matches snapshot with currentPage and totalPages prop assigned 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button class=\\"btn btn-default\\">
        1
      </button><button class=\\"btn btn-default\\">
        2
      </button><button class=\\"btn btn-default active\\">
        3
      </button><button class=\\"btn btn-default\\">
        4
      </button><button class=\\"btn btn-default\\">
        5
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button class=\\"btn btn-default\\">
        10
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 3
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;

exports[`Paginator.vue matches snapshot with currentPage, threshold, loading, and totalPages prop assigned 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button disabled=\\"disabled\\" title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        1
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        13
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        14
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        15
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        16
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        17
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        18
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        19
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default active\\">
        20
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        21
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        22
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        23
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        24
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        25
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        26
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        27
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        100
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 20
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button disabled=\\"disabled\\" title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;

exports[`Paginator.vue matches snapshot with loading and totalPages prop assigned 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button disabled=\\"disabled\\" title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button disabled=\\"disabled\\" class=\\"btn btn-default active\\">
        1
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        2
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        3
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        4
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        5
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default\\">
        10
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 1
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button disabled=\\"disabled\\" title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;

exports[`Paginator.vue matches snapshot with no props assigned 1`] = `""`;

exports[`Paginator.vue matches snapshot with threshold and totalPages prop assigned 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button disabled=\\"disabled\\" title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button class=\\"btn btn-default active\\">
        1
      </button><button class=\\"btn btn-default\\">
        2
      </button><button class=\\"btn btn-default\\">
        3
      </button><button class=\\"btn btn-default\\">
        4
      </button><button class=\\"btn btn-default\\">
        5
      </button><button class=\\"btn btn-default\\">
        6
      </button><button class=\\"btn btn-default\\">
        7
      </button><button class=\\"btn btn-default\\">
        8
      </button><button class=\\"btn btn-default\\">
        9
      </button><button class=\\"btn btn-default\\">
        10
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button class=\\"btn btn-default\\">
        100
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 1
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;

exports[`Paginator.vue matches snapshot with totalPages prop assigned 1`] = `
"<nav data-id=\\"sds-paginator\\" aria-label=\\"Page navigation\\" class=\\"paginator\\">
  <div role=\\"toolbar\\" class=\\"btn-toolbar\\">
    <div class=\\"mr-2 btn-group\\"><button disabled=\\"disabled\\" title=\\"First\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M11 19l-7-7 7-7m8 14l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">First</span></button> <button disabled=\\"disabled\\" title=\\"Prev\\" class=\\"flex space-x-1 btn btn-default\\"><svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M15 19l-7-7 7-7\\"></path>
        </svg> <span class=\\"sr-only sm:not-sr-only\\">Prev</span></button></div>
    <div class=\\"hidden btn-group md:block\\"><button class=\\"btn btn-default active\\">
        1
      </button><button class=\\"btn btn-default\\">
        2
      </button><button class=\\"btn btn-default\\">
        3
      </button><button class=\\"btn btn-default\\">
        4
      </button><button class=\\"btn btn-default\\">
        5
      </button><button disabled=\\"disabled\\" class=\\"btn btn-default shadow-none border-transparent\\">
        ...
      </button><button class=\\"btn btn-default\\">
        10
      </button></div>
    <div class=\\"block mx-0 currentPageLabelContainer btn-group md:hidden\\"><span class=\\"inline-block px-3 pt-2 currentPageLabel\\">
        Page 1
      </span></div>
    <div class=\\"ml-2 btn-group\\"><button title=\\"Next\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Next</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M9 5l7 7-7 7\\"></path>
        </svg></button> <button title=\\"Last\\" class=\\"flex space-x-1 btn btn-default\\"><span class=\\"sr-only sm:not-sr-only\\">Last</span> <svg fill=\\"none\\" stroke=\\"currentColor\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"w-4 h-4 my-auto\\">
          <path stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"3\\" d=\\"M13 5l7 7-7 7M5 5l7 7-7 7\\"></path>
        </svg></button></div>
  </div>
</nav>"
`;
